scripts/e2e/e2e.sh

```diff
@@ -140,6 +156,86 @@ install_govc() {
    export PATH=${govc_bin}:$PATH
 }
 
+build_upload_deploy_ovf() { 
+   # build the ova
+   ova_revision="v0.0.1-debug"
+   get_random_str
+   get_bootstrap_ssh_pub
+   docker volume create ova-tmp
+   docker run -t --rm --privileged -v /dev:/dev \
+     -v ova-tmp:/go/src/sigs.k8s.io/cluster-api-provider-vsphere/installer/bin \
+     gcr.io/cnx-cluster-api/cluster-api-provider-vsphere-installer:"$1" \
+     ova-ci --ci-root-password "${random_str}" --ci-root-ssh-key "${bootstrap_ssh_pub}" \
+     --build-ova-revision "${ova_revision}"
+
+   mkdir -p ./bin
+   CID=$(docker run -d -v ova-tmp:/ova-tmp busybox true)
+   docker cp "${CID}":/ova-tmp/cluster-api-vsphere-"${ova_revision}".ova ./bin/
+   docker stop "${CID}"
+   docker rm "${CID}"
+   docker volume rm ova-tmp
+
+   if [ ! -e ./bin/cluster-api-vsphere-"${ova_revision}".ova ]; then
+      echo "file not exist"
+      exit 1
+   fi
+ 
+   # upload and deploy bootstrap VM
+   name_prefix="clusterapi-bootstrap"
+   bootstrap_vm_name="$name_prefix"-"$random_str"
+   bootstrap_vm_folder="clusterapi"
+   bootstrap_vm_rp="clusterapi"
+   bootstrap_vm_network="sddc-cgw-network-3"
+   library_name="clusterapi"
+   datastore_name="WorkloadDatastore"
+   
+   govc library.create -ds="${datastore_name}" "${library_name}"
+   govc library.ova "${library_name}" "./bin/cluster-api-vsphere-${ova_revision}.ova"
+   govc vcenter.deploy -ds="${datastore_name}" -folder="${bootstrap_vm_folder}" -pool="${bootstrap_vm_rp}" "${library_name}" "cluster-api-vsphere-${ova_revision}.ova"  "$bootstrap_vm_name"
+   #govc library.rm "${library_name}"
+  
+
+   # wait for bootstrap_vm_ip
+   bootstrap_vm=$(govc find / -type m -name "$bootstrap_vm_name")
+   govc vm.network.change -vm "${bootstrap_vm_name}" -net "${bootstrap_vm_network}" ethernet-0
+   govc vm.power -on "${bootstrap_vm}"
+   retry=10
+   bootstrap_vm_ip=""
+   until [ $bootstrap_vm_ip ]
```

Comment:
Is retry needed here?  `vm.ip` already waits by default, and you can change that time using the `-wait` flag.

Reply:
thanks, changed,